lib/fog/libvirt/requests/compute/list_volumes.rb

```diff
@@ -69,23 +69,31 @@ def get_volume filter = { }, raw = false
               return raw ? vol : volume_to_attributes(vol)
             end
           end
-          { }
+
+          nil
         end
       end
 
       class Mock
         def list_volumes(filters={ })
           vol1 = mock_volume 'vol1'
           vol2 = mock_volume 'vol2'
-          [vol1, vol2]
+          vols = [vol1, vol2]
+
+          if filters.keys.empty?
+            return vols
+          end
+
+          key = filters.keys.first
+          vols.select { |v| v[key] == filters[key] }
         end
 
         def mock_volume name
           {
               :pool_name   => 'vol.pool.name',
               :key         => 'vol.key',
-              :id          => 'vol.key',
-              :path        => 'vol.path',
+              :id          => "vol.#{name}",
+              :path        => "path/to/disk", # used by in mock_files/domain.xml
```

Comment:
one of the tests results in searching for a volume with the path set to the same value in the domain.xml mock_files, with filtering enabled, this seemed like the easiest solution.